### PR TITLE
Add example for EfficientNet and upgrade PyTorch in Dockerfile

### PR DESCRIPTION
**Pull request by ultmaster**

### CODE REVIEW

File: examples/trials/efficientnet/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+EfficientNet-PyTorch
```
Comment (xuehui1991) [inline examples/trials/efficientnet/.gitignore]: ?

File: examples/trials/efficientnet/config_lr.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+maxExecDuration: 48h
+maxTrialNum: 140
+trainingServicePlatform: pai
+searchSpacePath: search_lr.json
+useAnnotation: false
+tuner:
+  builtinTunerName: GridSearch
+trial:
+  codeDir: EfficientNet-PyTorch
+  command: python3 main.py cifar10 -j 4 --epochs 20 -a efficientnet-b0 --pretrained --num-classes 10 --batch-size 64 --optimizer sgd --request-from-nni
```
Comment (xuehui1991) [inline examples/trials/efficientnet/config_lr.yml]: where is this code(main.py)?

File: examples/trials/efficientnet/config_lr.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+authorName: v-yugzh
```
Comment (xuehui1991) [inline examples/trials/efficientnet/config_lr.yml]: remove your name.

File: examples/trials/efficientnet/config_lr.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  builtinTunerName: GridSearch
+trial:
+  codeDir: EfficientNet-PyTorch
+  command: python3 main.py cifar10 -j 4 --epochs 20 -a efficientnet-b0 --pretrained --num-classes 10 --batch-size 64 --optimizer sgd --request-from-nni
+  cpuNum: 4
+  memoryMB: 16384
+  gpuNum: 1
+  shmMB: 8192
+  virtualCluster: nni
+  image: ultmaster/nni:latest
```
Comment (xuehui1991) [inline examples/trials/efficientnet/config_lr.yml]: could you use the image from nni?

File: examples/trials/efficientnet/config_net.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+authorName: v-yugzh
```
Comment (xuehui1991) [inline examples/trials/efficientnet/config_net.yml]: same here.

File: examples/trials/efficientnet/config_net.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    product: 2
+trial:
+  codeDir: EfficientNet-PyTorch
+  command: sh train_imagenet.sh
+  cpuNum: 4
+  memoryMB: 25000
+  shmMB: 25000
+  gpuNum: 1
+  virtualCluster: nni
+  image: ultmaster/nni:latest
```
Comment (xuehui1991) [inline examples/trials/efficientnet/config_net.yml]: same here.

File: examples/trials/efficientnet/.gitignore (hunk 1 above)
Comment (ultmaster) [reply examples/trials/efficientnet/.gitignore]: EfficientNet-PyTorch is with Apache license and cannot be included into our repo. So user has to clone this repo ahead. Add it to gitignore to prevent this directory to be included in our repo.

File: examples/trials/efficientnet/config_lr.yml (hunk 2 above)
Comment (ultmaster) [reply examples/trials/efficientnet/config_lr.yml]: The code is in EfficientNet-PyTorch repo.

File: examples/trials/efficientnet/config_lr.yml (hunk 3 above)
Comment (ultmaster) [reply examples/trials/efficientnet/config_lr.yml]: fixed

File: examples/trials/efficientnet/config_lr.yml (hunk 4 above)
Comment (ultmaster) [reply examples/trials/efficientnet/config_lr.yml]: In that case, PyTorch in NNI official image needs to be upgraded to at least v1.1; and nfs-common and cifs-utils need to be installed. Refer to [Dockerfile](https://github.com/ultmaster/nni-dockerfile/blob/master/Dockerfile) for all the changes I've done to make it work.

File: examples/trials/efficientnet/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+* Search space and tuners for finding the best tuple (alpha, beta, gamma) for EfficientNet-B1 with
+  grid search, as discussed in Section 3.3 in [paper](https://arxiv.org/abs/1905.11946).
+* Search space for EfficientNet transfer learning on datasets other than ImageNet;
+  settings are borrowed from Appendix A.5 from 
+  "[Do Better ImageNet Models Transfer Better?](https://arxiv.org/abs/1805.08974)".
+
+## Instructions
+
+1. Clone our modified version of [EfficientNet-PyTorch](https://github.com/ultmaster/EfficientNet-PyTorch) in this directory.
```
Comment (xuehui1991) [inline examples/trials/efficientnet/README.md]: what's the difference between the modified version and the original version?

File: examples/trials/efficientnet/config_lr.yml (hunk 2 above)
Comment (xuehui1991) [reply examples/trials/efficientnet/config_lr.yml]: please make clear in docs..

File: examples/trials/efficientnet/README.md (hunk 7 above)
Comment (ultmaster) [reply examples/trials/efficientnet/README.md]: documented

**Comment by scarlett2018** [comment]
**TODO**: Update the NNI docker image to with Pytorch 1.1, run the IT and UT.

File: examples/trials/efficientnet/README.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  grid search, as discussed in Section 3.3 in [paper](https://arxiv.org/abs/1905.11946).
+* Search space for EfficientNet transfer learning on datasets other than ImageNet;
+  settings are borrowed from Appendix A.5 from 
+  "[Do Better ImageNet Models Transfer Better?](https://arxiv.org/abs/1805.08974)".
+
+## Instructions
+
+1. Clone our modified version of [EfficientNet-PyTorch](https://github.com/ultmaster/EfficientNet-PyTorch).
+   The modifications were done to adhere to the original [Tensorflow version](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet) 
+   as close as possible (including EMA, label smoothing and etc.); also added are the interactions with NNI.
```
Comment (xuehui1991) [inline examples/trials/efficientnet/README.md]: also added are the interactions with NNI？ （sorry I cannot understand...）

File: examples/trials/efficientnet/README.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+   Clone it into `EfficientNet-PyTorch`; then you will see files like `main.py`, `train_imagenet.sh`, as specified
+   in the configuration files.
```
Comment (xuehui1991) [inline examples/trials/efficientnet/README.md]: why separate this sentence into 2 lines?

Comment (ultmaster) [reply examples/trials/efficientnet/README.md]: https://about.gitlab.com/2016/10/11/wrapping-text/

File: examples/trials/efficientnet/README.md (hunk 8 above)
Comment (ultmaster) [reply examples/trials/efficientnet/README.md]: I've made it more clear.

File: examples/trials/efficientnet/README.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# EfficientNet
+
+[arXiv](https://arxiv.org/abs/1905.11946)
```
Comment (leckie-chn) [inline examples/trials/efficientnet/README.md]: could you please change `arxiv` to other name such as the name of this paper?

File: examples/trials/efficientnet/README.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+We provide here:
+
+* Search space and tuners for finding the best tuple (alpha, beta, gamma) for EfficientNet-B1 with grid search, as discussed in Section 3.3 in [paper](https://arxiv.org/abs/1905.11946).
+* Search space for EfficientNet transfer learning on datasets other than ImageNet; settings are borrowed from Appendix A.5 from "[Do Better ImageNet Models Transfer Better?](https://arxiv.org/abs/1805.08974)".
+
+## Instructions
+
+1. Clone our modified version of [EfficientNet-PyTorch](https://github.com/ultmaster/EfficientNet-PyTorch). The modifications were done to adhere to the original [Tensorflow version](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet) as close as possible (including EMA, label smoothing and etc.); also added are the part which gets parameters from tuner and reports intermediate/final results. Clone it into `EfficientNet-PyTorch`; then you will see files like `main.py`, `train_imagenet.sh`, as specified in the configuration files.
```
Comment (leckie-chn) [inline examples/trials/efficientnet/README.md]: suggest adding one simple instruction here: `git clone ...`

File: examples/trials/efficientnet/README.md [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+2. Resolve all the dependencies following the README.
+3. Run `nnictl create --config config_lr.yml` if you want to do transfer learning; or `nnictl create --config config_net.yml` if you want to find the best EfficientNet-B1. Adjust the training service (PAI/local/remote), batch size in the config files according to your environment.
+
+For training on ImageNet, you should read `EfficientNet-PyTorch/train_imagenet.sh`. Download ImageNet beforehand and extract it adhering to [PyTorch format](https://pytorch.org/docs/stable/torchvision/datasets.html#imagenet) and then replace `/mnt/data/imagenet` in with the location of your ImageNet storage. This file should also be a good example in case you want to mount ImageNet into the container on OpenPAI.
```
Comment (leckie-chn) [inline examples/trials/efficientnet/README.md]: generally speaking, you should provide several lines of simple instructions so that users can just copy your code and run :)

File: examples/trials/efficientnet/tuner.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
+from nni.gridsearch_tuner.gridsearch_tuner import GridSearchTuner
+
+
+class FixedProductTuner(GridSearchTuner):
+
+    def __init__(self, product):
+        super().__init__()
+        self.product = product
+
+    def expand_parameters(self, para):
```
Comment (leckie-chn) [inline examples/trials/efficientnet/tuner.py]: better adding some docstring here?

File: examples/trials/efficientnet/README.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+2. Run `nnictl create --config config_lr.yml` if you want to do transfer learning; or `nnictl create --config config_net.yml` if you want to find the best EfficientNet-B1. Adjust the training service (PAI/local/remote), batch size in the config files according to your environment.
+
+For training on ImageNet, you should read `EfficientNet-PyTorch/train_imagenet.sh`. Download ImageNet beforehand and extract it adhering to [PyTorch format](https://pytorch.org/docs/stable/torchvision/datasets.html#imagenet) and then replace `/mnt/data/imagenet` in with the location of your ImageNet storage. This file should also be a good example in case you want to mount ImageNet into the container on OpenPAI.
```
Comment (QuanluZhang) [inline examples/trials/efficientnet/README.md]: “**This file** should also  be a good ...”, which file?

File: examples/trials/efficientnet/README.md [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+2. Run `nnictl create --config config_lr.yml` in case of transfer learning; or `nnictl create --config config_net.yml` to find the best EfficientNet-B1. Adjust the training service (PAI/local/remote), batch size in the config files according to the environment.
+
+For training on ImageNet, read `EfficientNet-PyTorch/train_imagenet.sh`. Download ImageNet beforehand and extract it adhering to [PyTorch format](https://pytorch.org/docs/stable/torchvision/datasets.html#imagenet) and then replace `/mnt/data/imagenet` in with the location of the ImageNet storage. This file should also be a good example to follow for mounting ImageNet into the container on OpenPAI.
+
+*Backward compatibility instructions: Grid Search quniform is changed in NNI v1.0. Use `search_net_old.json` if for NNI <= 0.9.*
```
Comment (chicm-ms) [inline examples/trials/efficientnet/README.md]: We should encourage customer to use latest version, I suggest to remove search_net_old.json

File: examples/trials/efficientnet/README.md [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+[EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks](https://arxiv.org/abs/1905.11946)
+
+Provided here are:
+
+* Search space and tuners for finding the best tuple (alpha, beta, gamma) for EfficientNet-B1 with grid search, as discussed in Section 3.3 in [paper](https://arxiv.org/abs/1905.11946).
+* Search space for EfficientNet transfer learning on datasets other than ImageNet; settings are borrowed from Appendix A.5 from "[Do Better ImageNet Models Transfer Better?](https://arxiv.org/abs/1805.08974)".
+
+## Instructions
+
+1. Run `git clone https://github.com/ultmaster/EfficientNet-PyTorch` to clone this modified version of [EfficientNet-PyTorch](https://github.com/lukemelas/EfficientNet-PyTorch). The modifications were done to adhere to the original [Tensorflow version](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet) as close as possible (including EMA, label smoothing and etc.); also added are the part which gets parameters from tuner and reports intermediate/final results. Clone it into `EfficientNet-PyTorch`; the files like `main.py`, `train_imagenet.sh` will appear inside, as specified in the configuration files.
```
Comment (chicm-ms) [inline examples/trials/efficientnet/README.md]: Since you added EfficientNet-PyTorch to .gitignore, it implies the efficient net repo need to be cloned into examples/trials/efficientnet.  So I think a ```cd xxx``` command should be added into this instruction before clone command.

File: deployment/docker/Dockerfile [hunk 17]
```diff
@@ -68,8 +68,8 @@ RUN python3 -m pip --no-cache-dir install Keras==2.1.6
 #
 # PyTorch
 #
-RUN python3 -m pip --no-cache-dir install torch==0.4.1
-RUN python3 -m pip install torchvision==0.2.1
+RUN python3 -m pip --no-cache-dir install torch==1.2.0
+RUN python3 -m pip install torchvision==0.4.0
```
Comment (chicm-ms) [inline deployment/docker/Dockerfile]: Suggest to use seperate PR to upgrade Dockfile

File: examples/trials/efficientnet/README.md (hunk 15 above)
Comment (ultmaster) [reply examples/trials/efficientnet/README.md]: I double-checked. NNI v1.0 dropped support for qloguniform in GridSearch. Using choice as workaround. Any suggestions?

File: deployment/docker/Dockerfile (hunk 17 above)
Comment (ultmaster) [reply deployment/docker/Dockerfile]: Technically this PR won't work without this Dockerfile update. Maybe I should update the title.

File: examples/trials/efficientnet/README.md (hunk 16 above)
Comment (ultmaster) [reply examples/trials/efficientnet/README.md]: I implied this in my README. Just add another step to make it clear.

File: examples/trials/efficientnet/README.md (hunk 15 above)
Comment (chicm-ms) [reply examples/trials/efficientnet/README.md]: If this example does not work with NNI v1.0, let's improve NNI first.   If choice works, it's fine to use choice instead.

File: deployment/docker/Dockerfile (hunk 17 above)
Comment (chicm-ms) [reply deployment/docker/Dockerfile]: it's OK for me to change PR name.

File: examples/trials/efficientnet/README.md (hunk 15 above)
Comment (QuanluZhang) [reply examples/trials/efficientnet/README.md]: @ultmaster you could use quniform, please ask @suiguoxin about how to use quniform in your scenario.

File: examples/trials/efficientnet/README.md [hunk 18]
```diff
@@ -0,0 +1,22 @@
+# EfficientNet
+
+[EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks](https://arxiv.org/abs/1905.11946)
+
+Provided here are:
+
+* Search space and tuners for finding the best tuple (alpha, beta, gamma) for EfficientNet-B1 with grid search, as discussed in Section 3.3 in [paper](https://arxiv.org/abs/1905.11946).
+* Search space for EfficientNet transfer learning on datasets other than ImageNet; settings are borrowed from Appendix A.5 from "[Do Better ImageNet Models Transfer Better?](https://arxiv.org/abs/1805.08974)".
+
```
Comment (chicm-ms) [inline examples/trials/efficientnet/README.md]: It seems the work in second paper (https://arxiv.org/abs/1805.08974)  is not specific to EfficientNet.   What we provided in this example for 2nd paper is just learning rate search (which we already have a lot of examples).
Can we remove the second part?  so that this exmaple will more concise and clearer.

File: examples/trials/efficientnet/README.md (hunk 15 above)
Comment (ultmaster) [reply examples/trials/efficientnet/README.md]: @suiguoxin Can you give some advice on what is the best practice to sample loguniform points in GridSearch?

File: examples/trials/efficientnet/README.md (hunk 18 above)
Comment (ultmaster) [reply examples/trials/efficientnet/README.md]: The 2nd paper is cited by EfficientNet, for the settings of transfer learning. And yes, it's not important in the big picture. The main reason I show this here is to provide an example on how to use transfer learning with EfficientNet-PyTorch. I'm fine with removing this part. What about @QuanluZhang ?

**Comment by ultmaster** [comment]
Referencing issue: #1487
